src/api-server-selector.js

```diff
@@ -314,15 +325,42 @@ export class ApiServerSelector extends EventsTargetMixin(AmfHelperMixin(LitEleme
     this.uri = uri;
   }
 
+  /**
+   * Retrieves custom base uris nodes assigned to the
+   * custom-base-uri slot
+   *
+   * @return {Array} Nodes assigned to custom-base-uri slot
+   */
+  _getExtraServers() {
+    const slot = this.shadowRoot.querySelector('slot[name="custom-base-uri"]');
+    return slot ? slot.assignedNodes() : [];
```

Comment:
1) The `assignedNodes()` returns a list of nodes and not element. This includes next nodes. You need to filter it by element nodes to enable proper processing. For example then you take this value and you check the `length` property. It will have different value then you expect.
2) This function returns Array and NodeList depending whether there are nodes or not. If you want an array then you need to use `Array.from()` on the result.